src/msg/Message.cc

```diff
@@ -234,7 +234,7 @@ void Message::encode(uint64_t features, int crcflags)
 
   footer.flags = CEPH_MSG_FOOTER_COMPLETE;
 
-  if (crcflags & MSG_CRC_DATA) {
+  if (!skip_header_crc && (crcflags & MSG_CRC_DATA)) {
     calc_data_crc();
```

Comment:
You've got a function parameter ```skip_header_crc``` but the CRC calculation you're actually skipping is for the ```data``` bufferlist, which doesn't do anything with the header crc. And I don't see ```calc_data_crc()``` getting invoked anywhere else so it looks to me like this PR just breaks checksumming?

Reply:
It's my fault, it should skip calc header crc.